activerecord/test/cases/insert_all_test.rb

```diff
@@ -136,4 +136,12 @@ def test_upsert_all_does_not_perform_an_upsert_if_a_partial_index_doesnt_apply
 
     assert_equal ["Out of the Silent Planet", "Perelandra"], Book.where(isbn: "1974522598").order(:name).pluck(:name)
   end
+
+  def test_insert_all_raises_on_unknown_attribute
+    skip unless supports_insert_on_duplicate_update?
```

Comment:
You don't need to skip this test for adapters that don't support `on_duplicate_update`

Just use `insert_all!` b/c every adapter supports vanilla bulk inserts, so:

```ruby
def test_insert_all_raises_on_unknown_attribute
  assert_raise ActiveRecord::UnknownAttributeError do
    Book.insert_all! [{ unknown_attribute: "Test" }]
  end
end
```

Reply:
fixed